twingate/client_connector_test.go

```diff
@@ -82,26 +102,46 @@ func TestClientConnectorCreateError(t *testing.T) {
 	  }
 	}`
 
-	r := ioutil.NopCloser(bytes.NewReader([]byte(createNetworkOkJson)))
-	client := createTestClient()
+		client := newHTTPMockClient()
+		defer httpmock.DeactivateAndReset()
+		httpmock.RegisterResponder("POST", client.GraphqlServerURL,
+			func(req *http.Request) (*http.Response, error) {
+				resp := httpmock.NewStringResponse(200, createNetworkOkJson)
+				return resp, errors.New("error_1")
+			})
+		remoteNetworkID := graphql.ID("test")
+
+		remoteNetwork, err := client.createConnector(remoteNetworkID)
+
+		assert.EqualError(t, err, "failed to create connector with id : Post \""+client.GraphqlServerURL+"\": error_1")
+		assert.Nil(t, remoteNetwork)
+	})
+}
+
+func TestClientConnectorEmptyCreateError(t *testing.T) {
+	t.Run("Test Twingate Resource : Client Connector Empty Create Error", func(t *testing.T) {
 
-	GetDoFunc = func(req *retryablehttp.Request) (*http.Response, error) {
-		return &http.Response{
-			StatusCode: 200,
-			Body:       r,
-		}, nil
-	}
-	remoteNetworkName := "test"
+		// response JSON
+		createNetworkOkJson := `{}`
 
-	remoteNetwork, err := client.createConnector(remoteNetworkName)
+		client := newHTTPMockClient()
+		defer httpmock.DeactivateAndReset()
+		httpmock.RegisterResponder("POST", client.GraphqlServerURL,
+			httpmock.NewStringResponder(200, createNetworkOkJson))
+		remoteNetworkID := graphql.ID(nil)
 
-	assert.EqualError(t, err, "failed to create connector: error_1")
-	assert.Nil(t, remoteNetwork)
+		remoteNetwork, err := client.createConnector(remoteNetworkID)
+
+		assert.EqualError(t, err, NewAPIErrorWithID(ErrGraphqlIDIsEmpty, "create", remoteNetworkResourceName, "remoteNetworkID").Error())
```

Comment:
please don't compare error with error compare it with the expected string like we do on other tests